diff --git a/src/widgets/bar/lv_bar.c b/src/widgets/bar/lv_bar.c
index 6fc7b23..4da2483 100644
--- a/src/widgets/bar/lv_bar.c
+++ b/src/widgets/bar/lv_bar.c
@@ -68,6 +68,9 @@ const lv_obj_class_t lv_bar_class = {
     .width_def = LV_DPI_DEF * 2,
     .height_def = LV_DPI_DEF / 10,
     .instance_size = sizeof(lv_bar_t),
+    .editable = LV_OBJ_CLASS_EDITABLE_INHERIT,
+    .group_def = LV_OBJ_CLASS_GROUP_DEF_INHERIT,
+    .theme_inheritable = LV_OBJ_CLASS_THEME_INHERITABLE_FALSE,
     .base_class = &lv_obj_class,
     .name = "bar",
 };
diff --git a/src/widgets/button/lv_button.c b/src/widgets/button/lv_button.c
index 096dc29..8a1ea32 100644
--- a/src/widgets/button/lv_button.c
+++ b/src/widgets/button/lv_button.c
@@ -34,6 +34,8 @@ const lv_obj_class_t lv_button_class  = {
     .height_def = LV_SIZE_CONTENT,
     .group_def = LV_OBJ_CLASS_GROUP_DEF_TRUE,
     .instance_size = sizeof(lv_button_t),
+    .editable = LV_OBJ_CLASS_EDITABLE_INHERIT,
+    .theme_inheritable = LV_OBJ_CLASS_THEME_INHERITABLE_FALSE,
     .base_class = &lv_obj_class,
     .name = "btn",
 };
diff --git a/src/widgets/image/lv_image.c b/src/widgets/image/lv_image.c
index 5fdcc12..769d8cb 100644
--- a/src/widgets/image/lv_image.c
+++ b/src/widgets/image/lv_image.c
@@ -109,6 +109,9 @@ const lv_obj_class_t lv_image_class = {
     .width_def = LV_SIZE_CONTENT,
     .height_def = LV_SIZE_CONTENT,
     .instance_size = sizeof(lv_image_t),
+    .editable = LV_OBJ_CLASS_EDITABLE_INHERIT,
+    .group_def = LV_OBJ_CLASS_GROUP_DEF_INHERIT,
+    .theme_inheritable = LV_OBJ_CLASS_THEME_INHERITABLE_FALSE,
     .base_class = &lv_obj_class,
     .name = "image",
 #if LV_USE_OBJ_PROPERTY
diff --git a/src/widgets/label/lv_label.c b/src/widgets/label/lv_label.c
index ae07112..a3f4f9a 100644
--- a/src/widgets/label/lv_label.c
+++ b/src/widgets/label/lv_label.c
@@ -96,6 +96,9 @@ const lv_obj_class_t lv_label_class = {
     .width_def = LV_SIZE_CONTENT,
     .height_def = LV_SIZE_CONTENT,
     .instance_size = sizeof(lv_label_t),
+    .editable = LV_OBJ_CLASS_EDITABLE_INHERIT,
+    .group_def = LV_OBJ_CLASS_GROUP_DEF_INHERIT,
+    .theme_inheritable = LV_OBJ_CLASS_THEME_INHERITABLE_FALSE,
     .base_class = &lv_obj_class,
     .name = "label",
 #if LV_USE_OBJ_PROPERTY
